TombEngine/Game/collision/floordata.cpp

```diff
@@ -924,4 +926,76 @@ namespace TEN::Collision::Floordata
 
 		return false;
 	}
+
+	// Draws color-coded spheres near the player describing collision block flags.
+	void DrawNearbyTileFlags(const ItemInfo& item)
+	{
+		constexpr auto DRAW_RANGE = BLOCK(3);
+		constexpr auto SPHERE_RADIUS = 64.0f;
+		auto debugTargetPage = RENDERER_DEBUG_PAGE::LOGIC_STATS;
+
+		auto point = GameVector(item.Pose.Position, item.RoomNumber);
+		auto pointColl = GetCollision(point);
+		const auto& room = g_Level.Rooms[point.RoomNumber];
+
+		// To optimize the process, only check tiles in player room and within a certain range.
+		int minX = std::max(item.Pose.Position.x - DRAW_RANGE, room.x) / BLOCK(1);
+		int maxX = std::min(item.Pose.Position.x + DRAW_RANGE, room.x + (room.xSize * BLOCK(1))) / BLOCK(1);
+		int minZ = std::max(item.Pose.Position.z - DRAW_RANGE, room.z) / BLOCK(1);
+		int maxZ = std::min(item.Pose.Position.z + DRAW_RANGE, room.z + (room.zSize * BLOCK(1))) / BLOCK(1);
+
+		for (int x = minX; x < maxX; x++)
+		{
+			for (int z = minZ; z < maxZ; z++)
+			{
+				point.x = BLOCK(x);
+				point.z = BLOCK(z);
+
+				pointColl = GetCollision(point);
+				point.y = pointColl.Position.Floor;
+
+				if (pointColl.Block->Stopper)
```

Comment:
would this be better in a switch-case statement?

Reply:
The issue is that a sector may have multiples flags activated at the same time. (monkey + stopper for example) so it would draw two sphers (the red and orange one).

If I put it like a switch, it would draw only 1, the first that it find the condition.

Reply:
Ah ok I understand :)